src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityMultiFurnace.java

```diff
@@ -98,27 +98,35 @@ protected void trySearchNewRecipe() {
             Recipe currentRecipe = null;
             IItemHandlerModifiable importInventory = getInputInventory();
             IMultipleTankHandler importFluids = getInputTank();
-            boolean dirty = checkRecipeInputsDirty(importInventory, importFluids);
+
             //inverse of logic in normal AbstractRecipeLogic
             //for MultiSmelter, we can reuse previous recipe if inputs didn't change
             //otherwise, we need to recompute it for new ingredients
             //but technically, it means we can cache multi smelter recipe, but changing inputs have more priority
-            if(dirty || forceRecipeRecheck) {
-                this.forceRecipeRecheck = false;
-                //else, try searching new recipe for given inputs
+            if (metaTileEntity.isInputsDirty()) {
+                metaTileEntity.setInputsDirty(false);
+                //Inputs changed, try searching new recipe for given inputs
                 currentRecipe = findRecipe(maxVoltage, importInventory, importFluids);
-                if (currentRecipe != null) {
-                    this.previousRecipe = currentRecipe;
-                }
             } else if (previousRecipe != null && previousRecipe.matches(false, importInventory, importFluids)) {
                 //if previous recipe still matches inputs, try to use it
                 currentRecipe = previousRecipe;
+            } else {
+                metaTileEntity.setInputsDirty(false);
+                currentRecipe = findRecipe(maxVoltage, importInventory, importFluids);
```

Comment:
This logic is essentially duplicated on either side of the check of the previous recipe. It should most likely be on one side or another.

Reply:
solved by initializing inputsDirty to true, otherwise multiblock would get stuck if saved with output full until you changed something in the inputs